sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/subscribe/subscriptionCheckboxFactory.groovy

```diff
@@ -0,0 +1,54 @@
+package life.qbic.portal.sampletracking.components.projectoverview.subscribe
+
+import com.vaadin.ui.CheckBox
+import life.qbic.business.project.subscribe.Subscriber
+import life.qbic.portal.sampletracking.components.projectoverview.ProjectSummary
+
+/**
+ * <b>Checkboxes to handle project subscription</b>
+ *
+ * <p>Creates a checkbox for a {@link ProjectSummary} and adds a listener that triggers unsubscription from the project</p>
+ *
+ * @since 1.0.0
+ */
+class subscriptionCheckboxFactory {
+
+    private final SubscribeProjectController subscribeProjectController
+    private final Subscriber subscriber
+
+    subscriptionCheckboxFactory(SubscribeProjectController subscribeProjectController, Subscriber subscriber){
+        this.subscribeProjectController = subscribeProjectController
+        this.subscriber = subscriber
+    }
+
+    /**
+     * Creates an instance of a {@link CheckBox} and adds a listener that triggers (un)subscription based on the checkbox value
+     * @param project A {@link ProjectSummary} to which the checkbox is bound
+     * @return A checkbox which is already preconfigured with listeners and bound to the subscribe use case
+     */
+    CheckBox createInstance(ProjectSummary project){
```

Comment:
Could also mean that an instance of the factory is created.
I would suggest renaming it to signal what is created. Furthermore, this can and should be static to ensure no side effects are caused in the instance of the factory.
```suggestion
    static CheckBox createSubscriptionCheckboxFor(ProjectSummary project) {
```

Reply:
I agree, you have a point here. What do you think about naming it `createSubscriptionCheckbox`?

Reply:
what about `getSubscriptionCheckbox(ProjectSummary projectSummary)`?